hr_payroll_analysis/models/hr_payslip.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding:utf-8 -*-
+from openerp import models, fields, api
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+    
+    analysis_line_ids = fields.One2many('hr.payslip.analysis.line', 
+                                        'payslip_id', 'Analysis Lines')
+
+    @api.multi
+    def process_sheet(self):
+        """
+        Create the analysis lines when the payslip is confirmed
+        """
+        super(HrPayslip, self).process_sheet()
+        self.remove_analysis_lines()
+        self.compute_analysis_lines()
+    
+    @api.multi
+    def remove_analysis_lines(self):
+        """
+        Make sure no analysis line has already been computed
+        """
+        for payslip in self:
+            for line in payslip.analysis_line_ids:
+                line.unlink()
```

Comment:
you may call directly self.mapped('analytic_line_ids').unlink() with no loop at all.